File: android/app/src/main/java/fi/haippirinki/l2p/Platform.java
package fi.haippirinki.l2p;

import android.app.Activity;
import android.content.Context;
import android.content.SharedPreferences;

public class Platform {
    private static SharedPreferences preferences = null;

    static void init(Activity activity) {
        preferences = activity.getPreferences(Context.MODE_PRIVATE);
    }

    static void deinit() {
        preferences = null;
    }

    public static boolean getBooleanPreference(String key) {
        return preferences.getBoolean(key, false);
    }

    public static float getFloatPreference(String key) {
        return preferences.getFloat(key, 0.f);
    }

    public static int getIntPreference(String key) {
        return preferences.getInt(key, 0);
    }

    public static long getLongPreference(String key) {
        return preferences.getLong(key, 0);
    }

    public static String getStringPreference(String key) {
        return preferences.getString(key, "");
    }

    public static void putBooleanPreference(String key, boolean value) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putBoolean(key, value);
        editor.apply();
    }

    public static void putFloatPreference(String key, float value) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putFloat(key, value);
        editor.apply();
    }

    public static void putIntPreference(String key, int value) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putInt(key, value);
        editor.apply();
    }

    public static void putLongPreference(String key, long value) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putLong(key, value);
        editor.apply();
    }

    public static void putStringPreference(String key, String value) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putString(key, value);
        editor.apply();
    }
}

File: win32/Platform.cpp
#include "../common/Platform.h"

#include <map>

namespace Platform
{
	std::map<std::string, bool> booleanPreferences;
	std::map<std::string, float> floatPreferences;
	std::map<std::string, int> intPreferences;
	std::map<std::string, long long> longPreferences;
	std::map<std::string, std::string> stringPreferences;

	bool getBooleanPreference(const std::string& key)
	{
		auto it = booleanPreferences.find(key);
		if(it != booleanPreferences.end())
		{
			return it->second;
		}
		return false;
	}

	float getFloatPreference(const std::string& key)
	{
		auto it = floatPreferences.find(key);
		if(it != floatPreferences.end())
		{
			return it->second;
		}
		return 0.f;
	}

	int getIntPreference(const std::string& key)
	{
		auto it = intPreferences.find(key);
		if(it != intPreferences.end())
		{
			return it->second;
		}
		return 0;
	}

	long long getLongPreference(const std::string& key)
	{
		auto it = longPreferences.find(key);
		if(it != longPreferences.end())
		{
			return it->second;
		}
		return 0;
	}

	std::string getStringPreference(const std::string& key)
	{
		auto it = stringPreferences.find(key);
		if(it != stringPreferences.end())
		{
			return it->second;
		}
		return "";
	}

	void putBooleanPreference(const std::string& key, bool value)
	{
		booleanPreferences[key] = value;
	}

	void putFloatPreference(const std::string& key, float value)
	{
		floatPreferences[key] = value;
	}

	void putIntPreference(const std::string& key, int value)
	{
		intPreferences[key] = value;
	}

	void putLongPreference(const std::string& key, long long value)
	{
		longPreferences[key] = value;
	}

	void putStringPreference(const std::string& key, const std::string& value)
	{
		stringPreferences[key] = value;
	}
}
